lib/ansible/modules/cloud/azure/azure_rm_image_facts.py

```diff
@@ -0,0 +1,232 @@
+            self.results['ansible_facts']['azure_images'] = self.list_images(self.name)
+        elif not self.name and self.resource_group:
+            self.results['ansible_facts']['azure_images'] = self.list_images_by_resource_group(self.resource_group)
+        elif not self.name and not self.resource_group:
+            self.results['ansible_facts']['azure_images'] = self.list_images()
+        return self.results
+
+    def get_image(self, resource_group, image_name):
+        '''
+        Returns image details based on its name
+        '''
+
+        self.log('Get properties for {0}'.format(self.name))
+
+        result = []
+        item = None
+        try:
+            item = self.compute_client.images.get(resource_group, image_name)
+        except CloudError as exc:
+            self.fail('Failed to list images - {0}'.format(str(exc)))
+
+        result = [self.format_item(item)]
+        return result
+
+    def list_images_by_resource_group(self, resource_group):
+        '''
+        Returns image details based on its resource group
+        '''
+
+        self.log('List images filtered by resource group')
+        response = None
+        try:
+            response = self.compute_client.images.list_by_resource_group(resource_group)
+        except CloudError as exc:
+            self.fail("Failed to list images: {0}".format(str(exc)))
+
+        return [self.format_item(x) for x in response if self.has_tags(x.tags, self.tags)] if response else []
+
+    def list_images(self, image_name=None):
+        '''
+        Returns image details in current subscription
+        '''
+
+        self.log('List images within current subscription')
+        response = None
+        results = []
+        try:
+            response = self.compute_client.images.list()
+        except CloudError as exc:
+            self.fail("Failed to list all images: {0}".format(str(exc)))
+
+        results = [self.format_item(x) for x in response if self.has_tags(x.tags, self.tags)] if response else []
+        if image_name:
+            results = [result for result in results if result['name'] == image_name]
+        return results
+
+    def format_item(self, item):
+        d = item.as_dict()
+        d = {
+            'resource_group': d['id'].split('/')[4],
+            'name': d['name'],
+            'location': d['location'],
+            'tags': d.get('tags'),
+            'source': d['source_virtual_machine']['id'] if 'source_virtual_machine' in d.keys() else None,
+            'os_type': d['storage_profile']['os_disk']['os_type'],
+            'os_state': d['storage_profile']['os_disk']['os_state'],
+            'os_disk_caching': d['storage_profile']['os_disk']['caching'],
+            'os_storage_account_type': d['storage_profile']['os_disk']['storage_account_type'],
+            'os_disk': d['storage_profile']['os_disk']['managed_disk']['id'] if 'managed_disk' in d['storage_profile']['os_disk'].keys() else None,
```

Comment:
Also as the os_disk_caching and other options?

Reply:
It seems os_disk_caching variable value is always set in response, from [None, ReadyOnly, ReadWrite]. So I think it is okay to skip the condition on it.